docs/report_stage_i.md

```diff
@@ -0,0 +1,497 @@
+
+In case individual measurements method is deemed unacceptable, following fallback plans are possible:
+  - for very cheap OPCODEs, substitute individual instruction occurrences (e.g. `ISZERO`) with sequences of many instructions with pre-balanced stack (e.g. `DUP1 x 99, ISZERO x 100`) and use the "measure batch" instrumentation,
+  - use a binomial or similar model to extract the duration of an event shorter than the timer resolution (**TODO ref to paper Platform Independent Timing of Java Virtual Machine Bytecode Instructions**),
+  - use CPU cycles as a proxy of computational cost, instead of wallclock duration.
+
+#### Timer overhead adjustment
+
+The reading of a timer has itself an overhead. In a sequence of steps:
+
+```
+t0 = timer.now()
+measured_code...
+t1 = timer.now()
+```
+
+what is really measured is both the `measured_code` and `timer.now()`, latter being the overhead of the timer.
+Since in our case for **individual instructions** measurements, `measured_code` sometimes executes within a few nanoseconds, it is necessary to take extra care to ensure that the timer overhead is as small as possible, and to account for it in the analysis step.
+
+We do that within our [preliminary results](https://htmlpreview.github.io/?https://github.com/imapp-pl/gas-cost-estimator/blob/master/src/analysis/exploration.nb.html) by subtracting an estimate of the overhead for all environments (`Go/geth`, `C/evmone`, `Rust/OpenEthereum`).
+
+The [estimation of timer overhead is done in these preliminary results](https://htmlpreview.github.io/?https://github.com/imapp-pl/gas-cost-estimator/blob/master/src/analysis/exploration_timers.nb.html).
+As the timer overhead estimate we use the minimum observed duration between two `timer.now()`s, one immediately after the other.
+
+Moving forward, we are planning to continue adjusting for the timer overhead, with the following improvements:
+- we'll monitor and register the timer overhead during the OPCODE measurements. This is motivated by the observation from the preliminary results, that the timer overhead measured exhibits periodic bumps. We will consider filtering out the measurements where the measure timer overhead was above a given threshold,
+- we should also separately conduct a statistical test that for every OPCODE, a measurement of the timer overhead is significantly smaller than the timer overhead, or prepare a similar argument **TODO ref validation section**.
+
+#### Repetition and accounting for warm-up
+
+We will repeat the execution a number of times, within the same instance of the EVM/eWASM environment (e.g. OS process etc.).
+Every such repetition will be called a **run**.
+
+Measurements for initial runs have been detected to be larger, and they might either be discarded or kept.
+
+It will be explored, whether normal node operation exhibits such "initial conditions" regularly or not.
+Regardless, all OPCODEs should receive fair treatment in this aspect.
```

Comment:
Again, caching influence should be discussed (e.g. does it make sense to force a cache flush before each run? or conversely, force cache fill?)